File: Typecast/TCResourceHeader.m
//
//  TCResourceHeader.m
//  Type Designer
//
//  Created by David Schweinsberg on 28/12/12.
//  Copyright (c) 2012 David Schweinsberg. All rights reserved.
//

#import "TCResourceHeader.h"
#import "TCDataInput.h"

@implementation TCResourceHeader

- (id)initWithDataInput:(TCDataInput *)dataInput
{
    self = [super init];
    if (self)
    {
        _dataOffset = [dataInput readInt];
        _mapOffset = [dataInput readInt];
        _dataLength = [dataInput readInt];
        _mapLength = [dataInput readInt];
    }
    return self;
}

@end

File: Typecast/TCResourceHeader.swift
//
//  TCResourceHeader.swift
//  Type Designer
//
//  Created by David Schweinsberg on 7/27/17.
//  Copyright © 2017 David Schweinsberg. All rights reserved.
//

import Foundation

class TCResourceHeader {
  let dataOffset: Int
  let mapOffset: Int
  let dataLength: Int
  let mapLength: Int

  init(dataInput: TCDataInput) {
    dataOffset = Int(dataInput.readInt())
    mapOffset = Int(dataInput.readInt())
    dataLength = Int(dataInput.readInt())
    mapLength = Int(dataInput.readInt())
  }
}
